Chaos-controller is crashloopbackoff after old IO configmap is applied

## Bug Report

**What version of Kubernetes are you using?**
<!-- You can run `kubectl version` -->
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"archive", BuildDate:"1970-01-01T00:00:00Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"$Format:%H$", GitTreeState:"archive", BuildDate:"1970-01-01T00:00:00Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}

**What did you do?**
<!-- If possible, provide a recipe for reproducing the error. How you installed chaos-mesh. -->
After old configmap is applied, the status of chaos-controller is crashloopbackoff.
The log is as follow:
`Controller manager Version: version.Info{GitVersion:"v0.0.0-master+$Format:%h$", GitCommit:"af17a7427e40b6dc7235921995b3917d36542c82", GitTreeState:"clean", BuildDate:"2020-06-03T05:23:27Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
2020-06-04T02:21:28.323Z	INFO	controller-runtime.metrics	metrics server is starting to listen	{"addr": ":10080"}
2020-06-04T02:21:28.324Z	INFO	controller-runtime.builder	Registering a mutating webhook	{"GVK": "pingcap.com/v1alpha1, Kind=PodChaos", "path": "/mutate-pingcap-com-v1alpha1-podchaos"}
2020-06-04T02:21:28.324Z	INFO	controller-runtime.webhook	registering webhook	{"path": "/mutate-pingcap-com-v1alpha1-podchaos"}
2020-06-04T02:21:28.324Z	INFO	controller-runtime.builder	Registering a validating webhook	{"GVK": "pingcap.com/v1alpha1, Kind=PodChaos", "path": "/validate-pingcap-com-v1alpha1-podchaos"}
2020-06-04T02:21:28.324Z	INFO	controller-runtime.webhook	registering webhook	{"path": "/validate-pingcap-com-v1alpha1-podchaos"}
2020-06-04T02:21:28.324Z	INFO	controller-runtime.builder	Registering a mutating webhook	{"GVK": "pingcap.com/v1alpha1, Kind=NetworkChaos", "path": "/mutate-pingcap-com-v1alpha1-networkchaos"}
2020-06-04T02:21:28.324Z	INFO	controller-runtime.webhook	registering webhook	{"path": "/mutate-pingcap-com-v1alpha1-networkchaos"}
2020-06-04T02:21:28.324Z	INFO	controller-runtime.builder	Registering a validating webhook	{"GVK": "pingcap.com/v1alpha1, Kind=NetworkChaos", "path": "/validate-pingcap-com-v1alpha1-networkchaos"}
2020-06-04T02:21:28.324Z	INFO	controller-runtime.webhook	registering webhook	{"path": "/validate-pingcap-com-v1alpha1-networkchaos"}
2020-06-04T02:21:28.324Z	INFO	controller-runtime.builder	Registering a mutating webhook	{"GVK": "pingcap.com/v1alpha1, Kind=IoChaos", "path": "/mutate-pingcap-com-v1alpha1-iochaos"}
2020-06-04T02:21:28.324Z	INFO	controller-runtime.webhook	registering webhook	{"path": "/mutate-pingcap-com-v1alpha1-iochaos"}
2020-06-04T02:21:28.325Z	INFO	controller-runtime.builder	Registering a validating webhook	{"GVK": "pingcap.com/v1alpha1, Kind=IoChaos", "path": "/validate-pingcap-com-v1alpha1-iochaos"}
2020-06-04T02:21:28.325Z	INFO	controller-runtime.webhook	registering webhook	{"path": "/validate-pingcap-com-v1alpha1-iochaos"}
2020-06-04T02:21:28.325Z	INFO	controller-runtime.builder	Registering a mutating webhook	{"GVK": "pingcap.com/v1alpha1, Kind=TimeChaos", "path": "/mutate-pingcap-com-v1alpha1-timechaos"}
2020-06-04T02:21:28.325Z	INFO	controller-runtime.webhook	registering webhook	{"path": "/mutate-pingcap-com-v1alpha1-timechaos"}
2020-06-04T02:21:28.325Z	INFO	controller-runtime.builder	Registering a validating webhook	{"GVK": "pingcap.com/v1alpha1, Kind=TimeChaos", "path": "/validate-pingcap-com-v1alpha1-timechaos"}
2020-06-04T02:21:28.325Z	INFO	controller-runtime.webhook	registering webhook	{"path": "/validate-pingcap-com-v1alpha1-timechaos"}
2020-06-04T02:21:28.325Z	INFO	controller-runtime.builder	Registering a mutating webhook	{"GVK": "pingcap.com/v1alpha1, Kind=KernelChaos", "path": "/mutate-pingcap-com-v1alpha1-kernelchaos"}
2020-06-04T02:21:28.325Z	INFO	controller-runtime.webhook	registering webhook	{"path": "/mutate-pingcap-com-v1alpha1-kernelchaos"}
2020-06-04T02:21:28.325Z	INFO	controller-runtime.builder	Registering a validating webhook	{"GVK": "pingcap.com/v1alpha1, Kind=KernelChaos", "path": "/validate-pingcap-com-v1alpha1-kernelchaos"}
2020-06-04T02:21:28.325Z	INFO	controller-runtime.webhook	registering webhook	{"path": "/validate-pingcap-com-v1alpha1-kernelchaos"}
2020-06-04T02:21:28.325Z	INFO	controller-runtime.builder	Registering a mutating webhook	{"GVK": "pingcap.com/v1alpha1, Kind=StressChaos", "path": "/mutate-pingcap-com-v1alpha1-stresschaos"}
2020-06-04T02:21:28.325Z	INFO	controller-runtime.webhook	registering webhook	{"path": "/mutate-pingcap-com-v1alpha1-stresschaos"}
2020-06-04T02:21:28.325Z	INFO	controller-runtime.builder	Registering a validating webhook	{"GVK": "pingcap.com/v1alpha1, Kind=StressChaos", "path": "/validate-pingcap-com-v1alpha1-stresschaos"}
2020-06-04T02:21:28.325Z	INFO	controller-runtime.webhook	registering webhook	{"path": "/validate-pingcap-com-v1alpha1-stresschaos"}
2020-06-04T02:21:28.325Z	INFO	setup	Setting up webhook server
2020-06-04T02:21:28.325Z	INFO	inject-webhook	Inferred ConfigMap	{"namespace": "chaos-testing", "filepath": "/var/run/secrets/kubernetes.io/serviceaccount/namespace"}
2020-06-04T02:21:28.325Z	INFO	inject-webhook	Creating Kubernetes client to talk to the api-server
2020-06-04T02:21:28.327Z	INFO	inject-webhook	Created ConfigMap watcher	{"apiserver": "https://10.155.0.1:443", "namespaces": "chaos-testing", "template labels": {"app.kubernetes.io/component":"template"}, "config labels": {"app.kubernetes.io/component":"webhook"}}
2020-06-04T02:21:28.327Z	INFO	controller-runtime.webhook	registering webhook	{"path": "/inject-v1-pod"}
2020-06-04T02:21:28.327Z	INFO	setup	Starting manager
2020-06-04T02:21:28.327Z	INFO	setup	Launching watcher for ConfigMaps
2020-06-04T02:21:28.327Z	INFO	controller-runtime.webhook.webhooks	starting webhook server
2020-06-04T02:21:28.327Z	INFO	inject-webhook	Watching for ConfigMaps for changes	{"namespace": "chaos-testing", "labels": {"app.kubernetes.io/component":"webhook"}}
2020-06-04T02:21:28.327Z	INFO	controller-runtime.manager	starting metrics server	{"path": "/metrics"}
2020-06-04T02:21:28.328Z	INFO	controller-runtime.controller	Starting EventSource	{"controller": "networkchaos", "source": "kind source: /, Kind="}
2020-06-04T02:21:28.328Z	INFO	controller-runtime.controller	Starting EventSource	{"controller": "iochaos", "source": "kind source: /, Kind="}
2020-06-04T02:21:28.328Z	INFO	controller-runtime.controller	Starting EventSource	{"controller": "podchaos", "source": "kind source: /, Kind="}
2020-06-04T02:21:28.328Z	INFO	controller-runtime.controller	Starting EventSource	{"controller": "kernelchaos", "source": "kind source: /, Kind="}
2020-06-04T02:21:28.328Z	INFO	controller-runtime.controller	Starting EventSource	{"controller": "stresschaos", "source": "kind source: /, Kind="}
2020-06-04T02:21:28.328Z	INFO	controller-runtime.controller	Starting EventSource	{"controller": "timechaos", "source": "kind source: /, Kind="}
2020-06-04T02:21:28.329Z	INFO	controller-runtime.certwatcher	Updated current TLS certificate
2020-06-04T02:21:28.330Z	INFO	controller-runtime.webhook	serving webhook server	{"host": "", "port": 9443}
2020-06-04T02:21:28.330Z	INFO	controller-runtime.certwatcher	Starting certificate watcher
2020-06-04T02:21:28.428Z	INFO	controller-runtime.controller	Starting Controller	{"controller": "networkchaos"}
2020-06-04T02:21:28.428Z	INFO	controller-runtime.controller	Starting workers	{"controller": "networkchaos", "worker count": 1}
2020-06-04T02:21:28.428Z	INFO	controller-runtime.controller	Starting Controller	{"controller": "iochaos"}
2020-06-04T02:21:28.429Z	INFO	controller-runtime.controller	Starting workers	{"controller": "iochaos", "worker count": 1}
2020-06-04T02:21:28.429Z	INFO	controller-runtime.controller	Starting Controller	{"controller": "kernelchaos"}
2020-06-04T02:21:28.429Z	INFO	controller-runtime.controller	Starting workers	{"controller": "kernelchaos", "worker count": 1}
2020-06-04T02:21:28.429Z	INFO	controller-runtime.controller	Starting Controller	{"controller": "stresschaos"}
2020-06-04T02:21:28.429Z	INFO	controller-runtime.controller	Starting workers	{"controller": "stresschaos", "worker count": 1}
2020-06-04T02:21:28.430Z	INFO	controller-runtime.controller	Starting Controller	{"controller": "timechaos"}
2020-06-04T02:21:28.430Z	INFO	controller-runtime.controller	Starting workers	{"controller": "timechaos", "worker count": 1}
2020-06-04T02:21:28.431Z	INFO	controller-runtime.controller	Starting Controller	{"controller": "podchaos"}
2020-06-04T02:21:28.431Z	INFO	controller-runtime.controller	Starting workers	{"controller": "podchaos", "worker count": 1}
2020-06-04T02:24:07.098Z	INFO	setup	Triggering ConfigMap reconciliation
2020-06-04T02:24:07.098Z	INFO	inject-webhook	Fetching Template Configs...
2020-06-04T02:24:07.104Z	INFO	inject-webhook	Fetched templates	{"templates count": 0}
2020-06-04T02:24:07.104Z	INFO	inject-webhook	Fetching Configs...
2020-06-04T02:24:07.109Z	INFO	inject-webhook	Fetched configs	{"configs count": 1}
2020-06-04T02:24:07.110Z	ERROR	inject-webhook	failed to load template args	{"payload": "name: chaosfs-cockroachdb\nselector:\n  namespace: test\n  labelSelectors:\n    \"app.kubernetes.io/component\": \"cockroachdb\"\ninitContainers:\n- name: inject-scripts\n  image: pingcap/chaos-scripts:latest\n  imagePullpolicy: Always\n  command: [\"sh\", \"-c\", \"/scripts/init.sh -d /cockroach/cockroach-data/data -f /cockroach/cockroach-data/fuse-data\"]\ncontainers:\n- name: chaosfs\n  image: pingcap/chaos-fs:latest\n  imagePullpolicy: Always\n  ports:\n  - containerPort: 65534\n  securityContext:\n    privileged: true\n  command:\n    - /usr/local/bin/chaosfs\n    - -addr=:65534\n    - -pidfile=/tmp/fuse/pid\n    - -original=/cockroach/cockroach-data/fuse-data\n    - -mountpoint=/cockroach/cockroach-data/data\n  volumeMounts:\n    - name: datadir\n      mountPath: /cockroach/cockroach-data\n      mountPropagation: Bidirectional\nvolumeMounts:\n- name: datadir\n  mountPath: /cockroach/cockroach-data\n  mountPropagation: HostToContainer\n- name: scripts\n  mountPath: /tmp/scripts\n- name: fuse\n  mountPath: /tmp/fuse\nvolumes:\n- name: scripts\n  emptyDir: {}\n- name: fuse\n  emptyDir: {}\npostStart:\n  db:\n    command:\n      - /tmp/scripts/wait-fuse.sh\n", "error": "template field is required for template args config"}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
github.com/pingcap/chaos-mesh/pkg/webhook/config/watcher.(*K8sConfigMapWatcher).GetConfigs
	/src/pkg/webhook/config/watcher/watcher.go:313
github.com/pingcap/chaos-mesh/pkg/webhook/config/watcher.(*K8sConfigMapWatcher).GetInjectionConfigs
	/src/pkg/webhook/config/watcher/watcher.go:214
main.watchConfig.func1
	/src/cmd/controller-manager/main.go:276
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x13df416]

goroutine 417 [running]:
github.com/pingcap/chaos-mesh/pkg/webhook/config/watcher.(*K8sConfigMapWatcher).GetConfigs(0xc00051fc00, 0xc0005ce990, 0x0, 0x0, 0x0, 0x0)
	/src/pkg/webhook/config/watcher/watcher.go:315 +0xad6
github.com/pingcap/chaos-mesh/pkg/webhook/config/watcher.(*K8sConfigMapWatcher).GetInjectionConfigs(0xc00051fc00, 0x174ff97, 0x23, 0x0)
	/src/pkg/webhook/config/watcher/watcher.go:214 +0x6e
main.watchConfig.func1(0xc0003b0180, 0xc00051fc00, 0xc0007ecba0)
	/src/cmd/controller-manager/main.go:276 +0x199
created by main.watchConfig
	/src/cmd/controller-manager/main.go:239 +0x53
`

The configmap I used:
`apiVersion: v1
kind: ConfigMap
metadata:
  name: chaosfs-cockroachdb
  namespace: chaos-testing
  labels:
    app.kubernetes.io/component: webhook
data:
  chaosfs-cockroach.yaml: |
    name: chaosfs-cockroachdb
    selector:
      namespace: test
      labelSelectors:
        "app.kubernetes.io/component": "cockroachdb"
    initContainers:
    - name: inject-scripts
      image: pingcap/chaos-scripts:latest
      imagePullpolicy: Always
      command: ["sh", "-c", "/scripts/init.sh -d /cockroach/cockroach-data/data -f /cockroach/cockroach-data/fuse-data"]
    containers:
    - name: chaosfs
      image: pingcap/chaos-fs:latest
      imagePullpolicy: Always
      ports:
      - containerPort: 65534
      securityContext:
        privileged: true
      command:
        - /usr/local/bin/chaosfs
        - -addr=:65534
        - -pidfile=/tmp/fuse/pid
        - -original=/cockroach/cockroach-data/fuse-data
        - -mountpoint=/cockroach/cockroach-data/data
      volumeMounts:
        - name: datadir
          mountPath: /cockroach/cockroach-data
          mountPropagation: Bidirectional
    volumeMounts:
    - name: datadir
      mountPath: /cockroach/cockroach-data
      mountPropagation: HostToContainer
    - name: scripts
      mountPath: /tmp/scripts
    - name: fuse
      mountPath: /tmp/fuse
    volumes:
    - name: scripts
      emptyDir: {}
    - name: fuse
      emptyDir: {}
    postStart:
      db:
        command:
          - /tmp/scripts/wait-fuse.sh`

**What did you expect to see?**
Chaos-controller is running.
**What did you see instead?**
Chaos-controller is crashloopbackoff.

--- a/controllers/metrics/metrics.go
+++ b/controllers/metrics/metrics.go
@@ -33,7 +33,7 @@ type ChaosCollector struct {
 	ConfigTemplates     *prometheus.GaugeVec
 	InjectionConfigs    *prometheus.GaugeVec
 	TemplateNotExist    *prometheus.CounterVec
-	TemplateLoadError   *prometheus.CounterVec
+	TemplateLoadError   prometheus.Counter
 	ConfigNameDuplicate *prometheus.CounterVec
 	InjectRequired      *prometheus.CounterVec
 	Injections          *prometheus.CounterVec
@@ -67,10 +67,10 @@ func NewChaosCollector(store cache.Cache, registerer prometheus.Registerer) *Cha
 			Name: "chaos_mesh_config_name_duplicate_total",
 			Help: "Total number of config name duplication error",
 		}, []string{"namespace", "config"}),
-		TemplateLoadError: prometheus.NewCounterVec(prometheus.CounterOpts{
+		TemplateLoadError: prometheus.NewCounter(prometheus.CounterOpts{
 			Name: "chaos_mesh_template_load_failed_total",
 			Help: "Total number of failures when rendering config args to template",
-		}, []string{"namespace", "template", "config"}),
+		}),
 		InjectRequired: prometheus.NewCounterVec(prometheus.CounterOpts{
 			Name: "chaos_mesh_inject_required_total",
 			Help: "Total number of injections required",

--- a/pkg/webhook/config/watcher/watcher.go
+++ b/pkg/webhook/config/watcher/watcher.go
@@ -312,7 +312,7 @@ func (c *K8sConfigMapWatcher) GetConfigs() ([]*config.TemplateArgs, error) {
 			if err != nil {
 				log.Error(err, "failed to load template args", "payload", payload)
 				if c.metrics != nil {
-					c.metrics.TemplateLoadError.WithLabelValues(conf.Namespace, conf.Template, conf.Name).Inc()
+					c.metrics.TemplateLoadError.Inc()
 				}
 				continue
 			}

